app/models/report.rb

```diff
@@ -56,23 +39,9 @@ class Report < ActiveRecord::Base
   # with_changes
   scope :interesting, -> { where("status <> 0") }
 
-  # a method that save the report values (e.g. values from METRIC)
-  # it is not supported to edit status values after it has been written once.
-  def status=(st)
-    s = case st
-          when Integer, Fixnum
-            st
-          when Hash
-            ReportStatusCalculator.new(:counters => st).calculate
-          else
-            raise Foreman::Exception(N_('Unsupported report status format'))
-        end
-    write_attribute(:status, s)
-  end
-
   # extracts serialized metrics and keep them as a hash_with_indifferent_access
   def metrics
-    YAML.load(read_attribute(:metrics)).with_indifferent_access
+    YAML.load(read_attribute(:metrics)).with_indifferent_access rescue {}
```

Comment:
Shouldn't this check the value of metrics, rather than rescuing an exception?

Is there any test coverage?